llvm/include/llvm/Support/Windows/WindowsSupport.h

```diff
@@ -245,6 +245,12 @@ std::error_code GetCommandLineArguments(SmallVectorImpl<const char *> &Args,
 std::error_code widenPath(const Twine &Path8, SmallVectorImpl<wchar_t> &Path16,
                           size_t MaxPathLen = MAX_PATH);
 
+/// Dynamically retrieve an exported function from a DLL, while casting it to a
+/// known type. This avoids -Wcast-function-type-mismatch.
+template <typename T> T funcFromModule(HMODULE Mod, StringRef Name) {
+  return (T)(void *)::GetProcAddress(Mod, Name.data());
+}
```

Comment:
What about something like:

```c++
template <typename ReturnType, typename... Arguments>
auto funcFromModule(HMODULE hMod, StringRef Name) -> ReturnType ((*)(Arguments...)) {
  return reinterpret_cast<ReturnType (*)(Arguments...)>(::GetProcAddress(hMod, Name.data());
}
```

Reply:
You can't destructure a C function pointer like this, can you? This needs `std::remove_pointer_t<>` on the calling site?